wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/WorkspaceActivityChecker.java

```diff
@@ -101,19 +104,29 @@ void cleanup() {
 
   private void stopAllExpired() {
     try {
-      activityDao.findExpired(clock.millis()).forEach(this::stopExpiredQuietly);
+      activityDao
+          .findExpiredIdle(clock.millis())
+          .forEach(wsId -> stopExpiredQuietly(wsId, WORKSPACE_IDLE_TIMEOUT_EXCEEDED));
+      if (workspaceActivityManager.getRunTimeout() > 0) {
+        activityDao
+            .findExpiredRunTimeout(clock.millis(), workspaceActivityManager.getRunTimeout())
+            .forEach(
+                wsId -> {
+                  LOG.info("{} for workspace {}", WORKSPACE_RUN_TIMEOUT_EXCEEDED, wsId);
+                  stopExpiredQuietly(wsId, WORKSPACE_RUN_TIMEOUT_EXCEEDED);
```

Comment:
what if we add a parameter `reason` to `stopExpiredQuietly` then this lambda will look a bit smaller.?